### PR TITLE
Hubs/Scopes Merge 36 - Implement `CombinedScopeViewTest`

### PR DESCRIPTION
**Pull request by adinauer**
#skip-changelog

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt [hunk 1]
```diff
@@ -108,6 +177,874 @@ class CombinedScopeViewTest {
...
+        combined.setTransaction("aTransaction")
+
+        assertEquals(ScopeType.ISOLATION, fixture.options.defaultScopeType)
+        assertNull(fixture.scope.transactionName)
+        assertEquals("aTransaction", fixture.isolationScope.transactionName)
+        assertNull(fixture.globalScope.transactionName)
+    }
+
+    @Test
+    fun `prefers transaction andspan from current scope`() {
```
Comment (markushi) [inline sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt:284]:
```suggestion
    fun `prefers transaction and span from current scope`() {
```

Comment (markushi) [reply sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt:284]: `nit`: A few more similar typos are below as well

**Review by markushi — APPROVED** [review]
Nice!

File: sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt [hunk 2]
```diff
@@ -108,6 +177,874 @@ class CombinedScopeViewTest {
         assertEquals(fourth, eventProcessors.get(3))
```
Comment (lbloder) [inline sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt:177]: (l) we could use kotlin list access syntax here -> `eventProcessors[0]`

File: sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt [hunk 3]
```diff
@@ -108,6 +177,874 @@ class CombinedScopeViewTest {
...
+        combined.setTag("aTag", "aValue")
+
+        assertEquals(ScopeType.ISOLATION, fixture.options.defaultScopeType)
+        assertNull(fixture.scope.tags["aTag"])
+        assertEquals("aValue", fixture.isolationScope.tags["aTag"])
+        assertNull(fixture.globalScope.tags["aTag"])
+    }
+
+    @Test
+    fun `prefer scope value for tags with same key`() {
```
Comment (lbloder) [inline sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt:527]: could be `prefer current scope value for tags with same key`

File: sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt [hunk 4]
```diff
@@ -108,6 +177,874 @@ class CombinedScopeViewTest {
...
+        combined.setExtra("someExtra", "aValue")
+
+        assertEquals(ScopeType.ISOLATION, fixture.options.defaultScopeType)
+        assertNull(fixture.scope.extras["someExtra"])
+        assertEquals("aValue", fixture.isolationScope.extras["someExtra"])
+        assertNull(fixture.globalScope.extras["someExtra"])
+    }
+
+    @Test
+    fun `prefer scope value for extras with same key`() {
```
Comment (lbloder) [inline sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt:598]: could be `prefer current scope value for extras with same key`

File: sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt [hunk 5]
```diff
@@ -108,6 +177,874 @@ class CombinedScopeViewTest {
...
+//        val contexts = combined.contexts
+//        assertEquals("scopeValue", contexts["scopeContext"])
+//    }
+
+    // TODO [HSM] test all setContext methods
+
+    // TODO [HSM] fingerprint tests (discuss how it should behave first)
+
+    @Test
+    fun `combines attachments ƒrom all scopes`() {
```
Comment (lbloder) [inline sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt:662]: non-ascii char `ƒ` could be replaced with `f`

File: sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt [hunk 6]
```diff
@@ -108,6 +177,874 @@ class CombinedScopeViewTest {
...
+    }
+
+    @Test
+    fun `getSpecificScope(CURRENT) returns scope`() {
+        val combined = fixture.getSut(SentryOptions().also { it.defaultScopeType = ScopeType.ISOLATION })
+        assertSame(fixture.scope, combined.getSpecificScope(ScopeType.CURRENT))
+    }
+
+    @Test
+    fun `getSpecificScope(ISOLATION) returns scope`() {
```
Comment (lbloder) [inline sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt:807]: would `getSpecificScope(ISOLATION) returns isolation scope` make sense?

File: sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt [hunk 7]
```diff
@@ -108,6 +177,874 @@ class CombinedScopeViewTest {
...
+        val combined = fixture.getSut(SentryOptions().also { it.defaultScopeType = ScopeType.CURRENT })
+        assertSame(fixture.isolationScope, combined.getSpecificScope(ScopeType.ISOLATION))
+    }
+
+    @Test
+    fun `getSpecificScope(GLOBAL) returns scope`() {
```
Comment (lbloder) [inline sentry/src/test/java/io/sentry/CombinedScopeViewTest.kt:813]: Maybe `getSpecificScope(GLOBAL) returns global scope`

**Review by lbloder — APPROVED** [review]
LGTM 👍

**Pull request merged**